cds_ils/ldap/api.py

```diff
@@ -279,11 +289,26 @@ def update_users():
         return 0, 0, 0
 
     # create a map by employeeID for fast lookup
+    ldap_users_emails = set()
     ldap_users_map = {}
     for ldap_user in ldap_users:
-        ldap_person_id = ldap_user_get(ldap_user, "employeeID")
-        ldap_users_map[ldap_person_id] = ldap_user
+        if "mail" not in ldap_user:
+            _log_info(
+                log_uuid,
+                "missing_email",
+                dict(employee_id=ldap_user_get(ldap_user, "employeeID")),
+            )
+            continue
+        email = ldap_user_get_email(ldap_user)
+        if email not in ldap_users_emails:
```

Comment:
I don't understand the purpose of the `ldap_users_emails`. Is there any chance two ldap users to have the same email?

Reply:
there is a chance that ldap user does not have an email

Reply:
Good question; unfortunately yes there are a few ldap users with the same email (ie. [<bureau@stauffersa.ch>]()), and this is what triggers an error later on. These accounts are a result of a bad management of the users (they were manually created from what I heard) and can be safely ignored. We also observed that the email could be occasionally absent from the ldap user data, which is very unfortunate. There is no explanation to that and we agreed to ignore users with no email since these would eventually have one the next time `update_users` is called.